Registration

Maksimkarpeko opened the pull request:
https://graceful-pithivier-eed62f.netlify.app/

Maksimkarpeko committed:
feat: create component input and button and add style for this components

Maksimkarpeko committed:
feat:
-create RegistrationForm and add style this form.
-create SingForm and add style this form

Maksimkarpeko committed:
feat:finish with  forms
-update style
-add validation for inputs

Maksimkarpeko committed:
feat:complete registration form

Maksimkarpeko committed:
feat:last Edits

Kulgeyko93 commented on index.html:
```diff
@@ -4,10 +4,13 @@
     <div id="root"></div>
-    <script type="module" src="../../src/page/MainPage/main.tsx"></script>
+    <script type="module" src="/src/app/App.tsx"></script>
   </body>
```
зачем тут 2 импорта?

Maksimkarpeko replied:
Там только один импорт)

Kulgeyko93 commented on package.json:
```diff
@@ -11,9 +11,17 @@
     "react": "^19.1.0",
-    "react-dom": "^19.1.0"
+    "react-dom": "^19.1.0",
+    "react-hook-form": "^7.56.0",
```
formik   и react-hook-form для форм? зачем 2 библиотеки?

Maksimkarpeko replied:
Формик удалил

Kulgeyko93 commented on src/features/RegistrationForm/api/request.ts:
```diff
@@ -0,0 +1,49 @@
+export const getUsers = axios.get(`${MY_BACKEND_LINK}/users/me`, {
+	headers: {
+		Authorization: `Bearer ${'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOjE1LCJlbWFpbCI6InRlc3Q0QHRlc3Quc2QiLCJ1c2VybmFtZSI6InRlc3Q0IiwiaWF0IjoxNzQ1MzMzMTgxLCJleHAiOjE3NzY4NjkxODF9.qZz9OFwb4SKpjskpfZf6DJh6joGcRpuNdA5kA0fklBE'}`,
+	},
```
это вообще что такое? ты всех по 1 токену авторизовываешь?

Maksimkarpeko replied:
Удалино

Kulgeyko93 commented on src/features/RegistrationForm/api/request.ts:
```diff
@@ -0,0 +1,49 @@
+
+
+export const getUsers = axios.get(`${MY_BACKEND_LINK}/users/me`, {
+	headers: {
```
помотри как аксиос настраивается, axios instance

Kulgeyko93 commented on src/features/RegistrationForm/api/request.ts:
```diff
@@ -0,0 +1,49 @@
+};
+
+export const singIn = axios.post(`${MY_BACKEND_LINK}/auth/sign-in`, {
+	email: 'test4@test.sd',
```
такого не должно быть

Maksimkarpeko replied:
Удалил

Kulgeyko93 commented on src/features/RegistrationForm/api/request.ts:
```diff
@@ -0,0 +1,49 @@
+			}
+		})
+		.catch(err => {
+			if (err.response) {
```
можно написать лучше
if (...) {

throw new error
}

....

Maksimkarpeko replied:
Исправил

Kulgeyko93 commented on src/features/RegistrationForm/styles/RegistrationForm.module.css:
```diff
@@ -0,0 +1,26 @@
+form {
+	display: flex;
```
названия модульных стилей , через lower case 
![{53E47392-671D-4C0B-832F-86D1AA22B6BB}](https://github.com/user-attachments/assets/7adb1308-9df6-4bc5-86ba-97fe8401156b)

Maksimkarpeko replied:
Исправил

Kulgeyko93 commented on src/features/RegistrationForm/styles/RegistrationForm.module.css:
```diff
@@ -0,0 +1,26 @@
+	margin-top: 5%;
+}
+.InputEmail,
+.InputUserName,
```
классы с маленькой буквы

Maksimkarpeko replied:
Исправил

Kulgeyko93 commented on src/features/RegistrationForm/ui/RegistrationForm.tsx:
```diff
@@ -0,0 +1,125 @@
+import { Input } from 'antd';
+import { Controller, SubmitHandler } from 'react-hook-form';
+import { Link } from 'react-router-dom';
+import { Button } from '../../../shared/ui/Button';
```
алиасы настрой

Maksimkarpeko replied:
Настроил алиасы

Kulgeyko93 commented on src/features/RegistrationForm/ui/RegistrationForm.tsx:
```diff
@@ -0,0 +1,125 @@
+	const [success, setSuccess] = useState<boolean>(false);
+	const { control, handleSubmit } = useRegistrForm();
+	const onSubmit: SubmitHandler<IRegistration> = data => {
+		addUser(data, setSuccess, setError);
```
типизация data лучше (data: ...) =>

Maksimkarpeko replied:
А есть ли разница между тем что я указал в <> и том что я написал в (data:IRegistration) Просто там data будет принимать IRegistation? Но наверное это более удобно для людей

Kulgeyko93 commented on src/features/SingForm/ui/SingForm.tsx:
```diff
@@ -0,0 +1,38 @@
+	return (
+		<>
+			<form action="" onSubmit={(e) => {
+				e.preventDefault();
```
типизация  e

Maksimkarpeko replied:
Удалил

Kulgeyko93 commented on src/features/SingForm/ui/SingForm.tsx:
```diff
@@ -0,0 +1,38 @@
+		<>
+			<form action="" onSubmit={(e) => {
+				e.preventDefault();
+				console.log("");
```
console.log(""); зачем

Maksimkarpeko replied:
удалил

Kulgeyko93 commented on src/shared/ui/Button.tsx:
```diff
@@ -0,0 +1,12 @@
+import { ClickType } from '../types/types'
+import style from '../styles/ui/Button.module.css'
+export const Button:FC<ClickType> = ({click}) =>{
+	return (
```
типизация

Kulgeyko93 commented on src/shared/ui/Button.tsx:
```diff
@@ -0,0 +1,12 @@
+import {FC} from 'react'
+import { ClickType } from '../types/types'
```
![{6B0E04DB-B217-4427-94EE-2E59D7E18F19}](https://github.com/user-attachments/assets/4f396934-c2c1-4a5b-9f05-30570bfff860)
types/index.ts лушче

Maksimkarpeko replied:
Можно было ли переменовать types в lib?

Kulgeyko93 replied:
index.ts

Maksimkarpeko committed:
feat:  -uninstall formik
-I cleaned file request from hard code
-I changed promis.then on async await
-I changed name all style in lowerCase
-I create alias from the long name path
fix: -I fix bag associated with password

Maksimkarpeko committed:
feat:-add working authorization
-add alias from singIn

Maksimkarpeko committed:
feat:add sing in
-add start menu

Kulgeyko93 commented on vite.config.ts:
```diff
@@ -1,7 +1,18 @@
+      "@LoginType":path.resolve(__dirname,"src/features/SingForm/module/typeLogin.ts")
+    },
+	},
+});
```
настрой общие алиасы
пример

![{41A6BA21-3AD0-48E6-ADF4-C0C66A0987EC}](https://github.com/user-attachments/assets/b6676bfa-8315-4903-b10c-537c346cd5e3)

Maksimkarpeko replied:
Сделал

Kulgeyko93 commented on src/page/StartPage/StartPage.tsx:
```diff
@@ -0,0 +1,18 @@
+		<>
+			<div className={style.containerPage}>
+				<h2>Welcome to LinkUp</h2>
+				<Link to={'/Registration'} className={style.link}>
```
все пути в enum  и отуда забирай пути (это константы)

Maksimkarpeko replied:
Сделал

Kulgeyko93 commented on src/features/SingForm/api/request.ts:
```diff
@@ -0,0 +1,38 @@
+				setSuccess(false);
+				setError('something went wrong');
+			} else if (err.response.status == 401) {
+				console.log('Был не найден пользователь');
```
прими за правило, делать как можно меньше воложенность, поэтом это кусок на рефакторинг

Kulgeyko93 commented on src/features/RegistrationForm/api/request.ts:
```diff
@@ -0,0 +1,32 @@
+	try {
+		const res = await axios.post(`${MY_BACKEND_LINK}/auth/sign-up`, {
+			email: data.email,
+			username: data.userName,
```
изучи как аксиос настраивается
https://axios-http.com/docs/instance
https://axios-http.com/docs/interceptors

Maksimkarpeko replied:
Вроде сделал

Maksimkarpeko committed:
feat:-work on alias
-add enum for links

Maksimkarpeko committed:
feat: add setting for axios

Maksimkarpeko committed:
feat:add request for Login

Maksimkarpeko committed:
feat: update UI website

Maksimkarpeko committed:
feat: update UI

Maksimkarpeko committed:
fix: fix problem

Maksimkarpeko committed:
feat:update UI

Maksimkarpeko committed:
feat: update UI and adaptive for mobile
